### ISSUE TITLE
Inconsistent syntax on 1.1

### ISSUE THREAD

**Issue by Roger-luo** [JuliaLang/julia#31736]
MWE:

```julia
julia> option(;kwargs...) = kwargs
option (generic function with 1 method)

julia> command(x; doc="") = x
command (generic function with 1 method)

julia> command((option(a=1), option(;b=2)); doc="aa")
ERROR: syntax: invalid syntax ; b = 2

julia> command([option(a=1), option(;b=2)]; doc="aa")
2-element Array{Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol},A} where A,1}:
 Base.Iterators.Pairs(:a=>1)
 Base.Iterators.Pairs(:b=>2)
```

This is not consistent with 1.0, while `[..]` works

**Comment by fredrikekre** [comment]
Works on master so would be good to find the fix and backport it.

**Comment by JeffBezanson** [comment]
Fixed by 4eb63085f136868a2296ca28654e3f40cdd068e6.
